### PR TITLE
[GOVUKAPP-767] Add all topics screen

### PR DESCRIPTION
**Pull request by 1pretz1**
## Video preview

https://github.com/user-attachments/assets/ab65abf3-604e-4e16-a7a5-e513b645d676

### CODE REVIEW

File: Production/govuk_ios/ViewControllers/AllTopicsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+        view.backgroundColor = UIColor.govUK.fills.surfaceBackground
+        setupUI()
+        configureConstraints()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        navigationController?.setNavigationBarHidden(
```
Comment (batesyuk) [inline Production/govuk_ios/ViewControllers/AllTopicsViewController.swift:44]: do this on the navigation item please

File: Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import UIKit
+import Foundation
+
+class AllTopicsViewModel {
+    private let analyticsService: AnalyticsServiceInterface
+    let topicAction: (Topic) -> Void
+    var topics: [Topic] = []
```
Comment (batesyuk) [inline Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift:7]: private(set)?

File: Production/govuk_ios/Views/Topics/TopicTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import UIKit
+import Foundation
+
+class TopicTableViewCell: UITableViewCell {
+    var topicCard: UIView = UIView()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
```
Comment (batesyuk) [inline Production/govuk_ios/Views/Topics/TopicTableViewCell.swift:8]: remove?

**Review by batesyuk — CHANGES_REQUESTED** [review]

File: Production/govuk_ios/Views/Topics/TopicTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        contentView.backgroundColor = UIColor.govUK.fills.surfaceBackground
+
+        topicCard = TopicRowView(topic: topic)
+        addSubview(topicCard)
+        configureConstraints()
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+
```
Comment (raphaelolu) [inline Production/govuk_ios/Views/Topics/TopicTableViewCell.swift:22]: white space

File: Production/govuk_ios/ViewControllers/AllTopicsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                equalTo: view.safeAreaLayoutGuide.bottomAnchor
+            )
+        ])
+    }
+}
+
+extension AllTopicsViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: TopicTableViewCell = tableView.dequeue(indexPath: indexPath)
+
```
Comment (raphaelolu) [inline Production/govuk_ios/ViewControllers/AllTopicsViewController.swift:77]: white space

**Review by raphaelolu — CHANGES_REQUESTED** [review]

File: Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift (hunk 2 above)
Comment (1pretz1) [reply Production/govuk_ios/ViewModels/Topics/AllTopicsViewModel.swift]: Done

**Review by batesyuk — DISMISSED** [review]

**Review by raphaelolu — APPROVED** [review]

**Pull request merged**